src/audio/volume.c

```diff
@@ -551,8 +546,11 @@ static int volume_prepare(struct comp_dev *dev)
 	comp_set_period_bytes(sinkb->sink, dev->frames, &cd->sink_format,
 			      &sink_period_bytes);
 
-	/* rewrite params format for all downstream */
-	dev->params.frame_fmt = cd->sink_format;
+	/* Rewrite params format for this component to match the host side. */
+	if (dev->params.direction == SOF_IPC_STREAM_PLAYBACK)
+		dev->params.frame_fmt = cd->source_format;
+	else
+		dev->params.frame_fmt = cd->sink_format;
```

Comment:
@singalsu setting the frame_fmt in prepare() might be a little too late if the idea is to propagate it to the components upstream/downstream. prepare() is always called params

Reply:
@ranj063 It really depends on the situation. Volume reads source and sink formats from connected components, so format propagation in this case doesn't change anything. Other components can need the connected formats in their copy functions, so this will be long after prepare has been called.

Reply:
@tlauda yes you are right. thanks